pallets/nfts/src/features/create_delete_item.rs

```diff
@@ -254,6 +263,10 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 					}
 				}
 
+				if AccountBalance::<T, I>::get(collection, &details.owner) == 1 {
```

Comment:
If the account only has one item in the balance, hence, remove the account from `collection_details.item_holders`.